starlite/testing.py

```diff
@@ -280,6 +281,188 @@ def test_my_handler() -> None:
     )
 
 
+class RequestFactory:
+    def __init__(
+        self,
+        app: Starlite = Starlite(route_handlers=[]),
+        server: str = "test.org",
+        port: int = 3000,
+        root_path: str = "",
+    ):
+        self.app = app
+        self.server = server
+        self.port = port
+        self.root_path = root_path
+
+    def _create_scope(
```

Comment:
Likewise, can we have `create_send` & `create_receive` also for the user to test his own middlewares. Better to keep these 3 public. We do need APIs for creation of `scope`, `send` & `receive`.

Reply:
@infohash as per my understanding `Recieve` and `Send` are basically callables that take/return a `Message` instance. I'm not sure it's related to the request factory 🤔 

I probably missed something, can you please guide me on what you'd expect the methods to do?

Reply:
When a middleware is called, it takes `scope`, `send`, & `receive` as arguments. We need these 3 for the user to test his own middlewares.

Reply:
Yeah I know, but I'm not sure how it's related to the `RequestFactory` and how you expect to use those methods/properties with it. 
Can you give an example of how you'd test a hypothetical middleware with a `RequestFactory` and those methods/properties?

Reply:
I think it's out of scope of `RequestFactory`. I will raise a separate PR for this. Then `_create_scope` method here will be moved.

Reply:
ok sure. Please tag me on the PR, I'm curious what you have in mind 😃